### DISCUSSION TITLE
What's the current version situation of wgpu?

### DISCUSSION THREAD

**Discussion by John-Nagle** [emilk/egui#4736] [General]
- wgpu is at 0.20.1 on crates.io, but at 0.20.0 at [Github trunk.](https://github.com/gfx-rs/wgpu/blob/trunk/Cargo.toml)
- There was some problem with wgpu 0.20.0 vs egui, as I recall, and egui backed off to 0.19. But now the versions of egui on crates.io and Github trunk are both using wgpu 0.20.0. 
- So what's the current recommended version situation?

**Comment by YgorSouza** [comment]
Each minor version has its own branch, so trunk can contain the potentially breaking changes from the next minor version.

https://github.com/gfx-rs/wgpu/blob/v0.20/Cargo.toml#L48

egui downgraded to wgpu 0.19 while waiting for the 0.20.1 patch, but when it came out they changed the dependency back to 0.20.0, not 0.20.1. I think that was a mistake. There is a 0.20.2 release on GitHub as well, but it's not yet on crates.io for some reason. [Rerun](https://github.com/rerun-io/rerun/blob/59276f92762ed8adaeafe297ad27fcd169ed635c/Cargo.lock#L7242-L7243) is on 0.20.1, so I suppose that's the version to use for now.

**Reply by John-Nagle** [reply]
Thanks. It's 0.19 -> 0.20 that has the breaking changes, so I want to get past that transition.

**Reply by John-Nagle** [reply]
The latest version on crates.io, 0.27.2, is still using wgpu 0.19.1. From the Cargo.toml file there:

```
[dependencies.wgpu]
version = "0.19.1"
features = [
    "fragile-send-sync-non-atomic-wasm",
    "wgsl",
]
```

(Read from here: https://crates.io/api/v1/crates/egui-wgpu/0.27.2/download 
(The returned file is type .tar.gz))

**Reply by YgorSouza** [reply]
Yes, wgpu 0.20 wasn't available yet when the latest egui was released. And eframe exposes wgpu types, so it can't upgrade without breaking changes. I thought you meant you were tracking egui master. If you are on crates.io, you have to wait until the next major. There is no schedule, only a tentative [list](https://github.com/emilk/egui/milestone/3) of issues that should be closed before the release. But since it's already been 3 months, emilk might decide to release soon even if some of these are still open.

**Reply by John-Nagle** [reply]
> If you are on crates.io, you have to wait until the next major.

> [bug list](https://github.com/emilk/egui/milestone/3)

Right. Maybe in the fall.

Anyone successfully getting good results with the Github version of egui-winit? 

I'm trying to get past a problem in wgpu. The wgpu people don't want to hear about bugs in old versions of wgpu. Current wgpu requires a newer wgpu-egui, but the bug list indicates that using the development version of egui is buying into a new set of problems.

**Reply by emilk** [reply]
A new egui release is coming in the next week 🤞

**Reply by John-Nagle** [reply]
Excellent! Thank you!

**Comment by John-Nagle** [comment]
It's here, and it works. Thanks.
